### PR TITLE
cache frontpage

### PR DESCRIPTION
**Pull request by robertkowalski**

### CODE REVIEW

File: facets/submissions/index.js [hunk 1]
```diff
@@ -230,13 +231,15 @@ exports.register = function Submissions (facet, options, next) {
...
       }
+
       request.server.methods.saveSubmission(payload, function (err, doc) {
         created = getCreatedFromCookie(request);
         created.push(doc.id);
-
-        reply()
-          .redirect('/contributions/' + doc.id)
-          .state('created', {created: created});
+        request.server.methods.getContributionsAndInterests.cache.drop(function () {
```
Comment (robin-drexler) [inline facets/submissions/index.js:238]: Does calling `.cache.drop` without a key cause the entire cache (for this method) to be invalidated?

Regarding to https://github.com/hapijs/catbox, .cache.drop has the following signature `key, callback`, but key is missing here? That's why I'm asking.

**Comment by robin-drexler** [comment]
Lgtm. 
Just thought that we might should add some tests for it. Especially the cache dropping when a new submission is created, is an important thing, that should not unintentionally break. (but easily can)

File: facets/submissions/index.js (hunk 1 above)
Comment (robertkowalski) [inline facets/submissions/index.js:238]: yes as we are not using catbox when using the simple in-memory-adapter. docs say:

```
server.method() created a new policy with segment: '#sum' 
automatically for us. Also the unique item id (cache key) 
was automatically generated from parameters. 
```

then when you take a look how the cache works, you'll see that if you don't define a name, the key will be an empty string `''` for getting the cache and deleting it:

https://github.com/hapijs/hapi/blob/v8.0.0/lib/methods.js#L155

and

https://github.com/hapijs/hapi/blob/v8.0.0/lib/methods.js#L184-200

**Comment by robertkowalski** [comment]
good point. i added an explanation and an integration test

**Comment by robertkowalski** [comment]
mind taking a second look?

**Comment by robin-drexler** [comment]
Cool, lgtm :)

**Comment by robertkowalski** [comment]
merged as b69959bd15b1652bcf22765f9ac014b8b0c36097

**Pull request closed without merging**